src/materials/DamagePlasticityStressUpdate.C

```diff
@@ -0,0 +1,682 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+#include "DamagePlasticityStressUpdate.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("BlackBearApp", DamagePlasticityStressUpdate);
+
+InputParameters
+DamagePlasticityStressUpdate::validParams()
+{
+  InputParameters params = MultiParameterPlasticityStressUpdate::validParams();
+  params.addParam<Real>(
+      "yield_function_tolerance",
+      "If the yield function is less than this amount, the (stress, internal parameters) are "
+      "deemed admissible.  A std::vector of tolerances must be entered for the multi-surface case");
+  params.addRangeCheckedParam<Real>("factor_relating_biaxial_unixial_cmp_str",
+                                    0.1,
+                                    "factor_relating_biaxial_unixial_cmp_str < 0.5 & "
+                                    "factor_relating_biaxial_unixial_cmp_str >= 0",
+                                    "Material parameter that relate biaxial and uniaxial "
+                                    "compressive  strength, i.e., \alfa = (fb0-fc0)/(2*fb0-fc0)");
+  params.addRequiredParam<Real>("factor_controlling_dilatancy",
```

Comment:
I'd suggest changing this to `dilatancy_factor`

Reply:
Addressed